core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/ModelMetricsFullHistory.kt

```diff
@@ -0,0 +1,125 @@
+package org.evomaster.core.problem.rest.classifier
+
+import kotlin.math.sqrt
+
+/**
+ * Tracks the full history of classifier performance metrics,
+ * without any time window (cumulative statistics).
+ *
+ * Maintains confusion-matrix style counters for binary classification:
+ * - Positive class = HTTP 400 responses
+ * - Negative class = all other responses
+ *
+ * @property correctPrediction the number of correct predictions made by the classifier
+ * @property totalSentRequests the total number of requests evaluated
+ * @property truePositive400 number of times a 400 was correctly predicted (TP)
+ * @property falsePositive400 number of times 400 was predicted but not actually 400 (FP)
+ * @property falseNegative400 number of times 400 was the actual result but not predicted (FN)
+ * @property trueNegative400 number of times not-400 was correctly predicted (TN)
+ */
+class ModelMetricsFullHistory : ModelMetrics {
+
+    var correctPrediction: Int = 0
+        private set
+    var totalSentRequests: Int = 0
+        private set
+
+    var truePositive400: Int = 0
+        private set
+    var falsePositive400: Int = 0
+        private set
+    var falseNegative400: Int = 0
+        private set
+    var trueNegative400: Int = 0
+        private set
+
+    /**
+     * Accuracy = (TP + TN) / (TP + TN + FP + FN)
+     */
+    override fun estimateAccuracy(): Double {
+        return if (totalSentRequests > 0) {
+            correctPrediction.toDouble() / totalSentRequests
+        } else 0.0
+    }
+
+    /**
+     * Precision(400) = TP / (TP + FP)
+     *
+     * Of all requests predicted as 400, how many were truly 400.
+     */
+    override fun estimatePrecision400(): Double {
+        val denom = truePositive400 + falsePositive400
+        return if (denom > 0) truePositive400.toDouble() / denom else 0.0
+    }
+
+    /**
+     * Recall(400) = TP / (TP + FN)
+     *
+     * Of all requests that were actually 400, how many were correctly predicted as 400.
+     */
+    override fun estimateRecall400(): Double {
+        val denominator = truePositive400 + falseNegative400
+        return if (denominator > 0) truePositive400.toDouble() / denominator else 0.0
```

Comment:
why in a method it is called `denom` but here `denominator`? be consistent :)  as rule of thumb, I prefer full names, eg `denominator`, unless the code get too cluttered